pkg/util/annotations/annotations.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	"fmt"
+	"regexp"
+)
+
+const (
+	metadataAnnotationPattern string = "^metadata\\.annotations\\['(.+)'\\]$"
+)
+
+// AnnotationFieldPathParser is a utility which validates and parses annotation field paths.
+// The format of an annotation field path is: metadata.annotations['key']
+type AnnotationFieldPathParser struct {
+	annotationRegexp *regexp.Regexp
+}
+
+// NewAnnotationFieldPathParser returns a new annotation field path parser.
+func NewAnnotationFieldPathParser() *AnnotationFieldPathParser {
```

Comment:
Why is the a function that returns a type, rather than just a validate function?  From the callers it seems that they would all be simpler.